src/realm/alloc.hpp

```diff
@@ -52,6 +52,10 @@ class MemRef {
     ref_type get_ref();
     void set_ref(ref_type ref);
     void set_addr(char* addr);
+    explicit operator bool() const noexcept
+    {
+        return (m_addr != nullptr);
+    }
```

Comment:
If I read this correct, this parenthesis operator just returns whether the `MemRef` is actually pointing to something or not? I would suggest using a more explicit name and not overloading an operator for this.